spec/lib/manageiq/api/common/request_spec.rb

```diff
@@ -0,0 +1,47 @@
+describe ManageIQ::API::Common::Request do
+  let(:request_good) do
+    headers = ActionDispatch::Http::Headers.from_hash(:blah           => 'blah',
+                                                      'x-rh-identity' => 'abc')
+    ActionDispatch::Request.new(headers)
+  end
+  let(:request_bad) { {:blah => 'blah' } }
+
+  context "user" do
+    it "returns a lazy loaded user instance" do
+      described_class.with_request(request_good) do |instance|
+        expect(instance.user).to be_a(ManageIQ::API::Common::User)
+      end
+    end
+  end
+
+  context "set current headers" do
+    it 'sets if the class is correct' do
+      described_class.with_request(request_good) do |instance|
+        expect(instance).to be_a(ManageIQ::API::Common::Request)
+        expect(instance.headers).to be_a(ActionDispatch::Http::Headers)
+      end
+    end
+
+    it 'raises an exception if the class is incorrect' do
+      expect do
+        described_class.with_request(request_bad) {}
+      end.to raise_exception(ArgumentError)
+    end
+  end
+
+  context "get forwardable headers" do
```

Comment:
Can you make this match the method name under test?  i.e. `describe ".current_forwardable"`